src/src/components/Contexts/SettingsContext.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { ajax } from '../../utils/ajax';
+import { AuthContext } from './AuthContext';
+
+type UserSettings = {
+  readonly weekStart: number;
+};
+
+export function SettingsProvider({
+  children,
+}: {
+  readonly children: React.ReactNode;
+}): JSX.Element {
+  const authContext = React.useContext(AuthContext);
+  const [weekStart, setWeekStart] = React.useState<number>(0);
+  const [found, setFound] = React.useState<boolean>(false);
+
+  // If the token has been set, try an update
+  // Also make sure we havent already fetched
+  if (authContext.token && !found) {
```

Comment:
You should not be making effects outside of React.useEffect() - this is likely to lead to duplicate network requests

Reply:
Should be fixed now I believe